### ISSUE TITLE
Access token available through JWT session v2 vs v3 behavior

### ISSUE THREAD

**Issue by dmi3y** [nextauthjs/next-auth#457]
*Please refer to the [documentation](https://next-auth.js.org/getting-started/introduction), the [example project](https://github.com/iaincollins/next-auth-example) and existing issues before creating a new issue.*

Thank you for an awesome project, I have a question/comment in regards v2 vs v3 behavior, please let me know if I need to clarify it more.

**Your question**

How to store/retrieve the provider access token with v3 in order to use it for external API calls.

This is follow up on the [question](https://github.com/iaincollins/next-auth/pull/399#issuecomment-660348440) by @tsaiDavid

Provided [extended](https://github.com/dmi3y/next-auth-example) version of the example project to illustrate the issue. Look at two branches `v2` and `v3`.
Specifically, it uses the Auth0 provider. And printing out session result on protected page:

### V3 output 
<img width="710" alt="localhost_3000_protected_v2" src="https://user-images.githubusercontent.com/1277279/87992516-4f758700-ca9d-11ea-99bf-5670990434d0.png">

### V3 output
<img width="723" alt="localhost_3000_protected_v3" src="https://user-images.githubusercontent.com/1277279/87992692-a5e2c580-ca9d-11ea-8679-a07f21222ef1.png">

**What are you trying to do**
v2 allows you to expose access token (and other information) via session callback, like so:

```
    session: async (session, data) => {
      // Access token available in the data for version v2
      // https://github.com/iaincollins/next-auth/compare/main..v3#diff-06cbba295b0b2816cc9c0ff790ff8302R68
      session.custom = "custom session variable";
      session.account = data.account;
      return Promise.resolve(session);
    },
```

in v3 it is not available anymore and such the code above is not working anymore. I believe that [these](https://github.com/iaincollins/next-auth/compare/main..v3#diff-06cbba295b0b2816cc9c0ff790ff8302R68) changes (~L68) are resulting in this descreptency.

**Documentation feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [x] Found the example project helpful
* [ ] Did not find the example project helpful

PS: ~I'm trying to get it deployed somewhere for live examples, will update description when succeed.~

Live deployment for [v2](https://next-auth-example-v2.dmitrii1.vercel.app/) and [v3](https://next-auth-example-v3.dmitrii1.vercel.app/) use `buddy@example.com/password` credentials. Note: signout functionality somewhat broken.

**Comment by iaincollins** [comment]
Thanks for raising this and going into detail! That is super helpful.

I am not sure if this is a bug or a gap in our documentation for v3 but I'm looking at both and will update with more info shortly.

**Comment by iaincollins** [comment]
This should be resolved with [the new callbacks in version 3.0](https://next-auth.js.org/configuration/callbacks)!

The User object is now passed to the Session callback, this can be used to look up the configured OAuth Provider accounts for a user by querying the `accounts` table for the User ID.

If not using a database, the JWT payload is passed instead of a User object. The account data is available to the JWT, where it can also be stored and passed to the Session object, for those using NextAuth.js without a database.

**Comment by dmi3y** [comment]
Thank you!
